stdlib/array.mli

```diff
@@ -328,6 +328,17 @@ val fast_sort : ('a -> 'a -> int) -> 'a array -> unit
 (** Same as {!sort} or {!stable_sort}, whichever is
     faster on typical input. *)
 
+val shuffle :
+  rand: (* thwart tools/sync_stdlib_docs *) (int -> int) -> 'a array -> unit
```

Comment:
We could make the `rand` argument optional, and default to `Random.int`

Reply:
This would cause a circular dependency between Array and Random.  But it would be fine if this function was in Random.

Reply:
Personally I am fine with the current choice:
- I think that if this was in the Random module, it would be more natural to have `Random.shuffle : 'a array -> 'a array` and `Random.State.shuffle : Random.State.t -> 'a array -> 'a array`
- but I see benefits in abstracting over the random generator
- the convenience cost for the user is fairly small, and it forces them to wonder if they want to pass a specific state or rely on the implicit domain state

I see the abstraction over a sampling function as arguably fairly similar to abstracting over a comparison function for sorting functions.